Update proposals and eth oracle details (#767)
vegaprotocol/documentation: docs/tutorials/using-data-sources.md

```javascript
"dataSourceSpecForSettlementData": {
    "external": {
        "ethOracle": {
            "address": "0x1b44F3514812d835EB1BDB0acB33d3fA3351Ee43",
            "abi": "[{\"inputs\":[],\"name\":\"latestAnswer\",\"outputs\":[{\"internalType\":\"int256\",\"name\":\"\",\"type\":\"int256\"}],\"stateMutability\":\"view\",\"type\":\"function\"}]",
            "method": "latestAnswer",
            "normalisers": [{
                  "name": "btc.price",
                  "expression": "$[0]"
              }],
            "requiredConfirmations": 3,
            "trigger": {
                "timeTrigger": {
                    "every": 30
                }
            },
            "filters": [{
                    "key": {
                        "name": "btc.price",
                        "type": "TYPE_INTEGER",
                        "numberDecimalPlaces": 8
                    },
                    "conditions": [
                        {
                            "operator": "OPERATOR_GREATER_THAN_OR_EQUAL",
                            "value": "0"
                        }
                    ]
                }
            ]
        }
    }
}
```

> This will cause the network validators to read the result from the specified smart contract and submit the result to Vega. When the data is verified by enough validators, this price is accepted on to the network. This is unlike the other two Oracle types ([Open Oracle signed messages](#open-oracle-signed-messages) & [JSON signed message data](#json-signed-message-data)),which both rely on a third party submitting data to the network.

Reviewer: ```suggestion
This will cause the network validators to read the result from the specified smart contract and submit the result to Vega. When the data is verified by enough validators, this price is accepted on to the network. This is unlike the other two Oracle types - [Open Oracle signed messages](#open-oracle-signed-messages) and [JSON signed message data](#json-signed-message-data) - which both rely on a third party submitting data to the network.
```

### ABI

> The `address` field in the specification tells the spec above which address to interact with on Ethereum. The `abi` ([Application Binary Interface](https://docs.soliditylang.org/en/develop/abi-spec.html#json) & `method` field on the spec above tells the settlement spec **how** to interact with it. Ethereum Oracle settlement specifications use the JSON ABI of the smart contract to describe the method on the contract that will be called to fetch the data. The ABI will contain the function name, details of any paramters required, and the format of the response. 

Reviewer: ```suggestion
The `address` field in the specification tells the spec above which address to interact with on Ethereum. The `abi` ([Application Binary Interface ↗](https://docs.soliditylang.org/en/develop/abi-spec.html#json) and `method` field on the spec above tells the settlement spec **how** to interact with it. Ethereum oracle settlement specifications use the JSON ABI of the smart contract to describe the method on the contract that will be called to fetch the data. The ABI will contain the function name, details of any paramters required, and the format of the response. 
```

> For example, the [Chainlink BTC/USD oracle](https://data.chain.link/ethereum/mainnet/crypto-usd/btc-usd) has its JSON ABI [published on Etherscan](https://etherscan.io/address/0xf4030086522a5beea4988f8ca5b36dbc97bee88c#code). When defining the data source spec, you can populate the `abi` field with the full ABI, and then set the `method` to `latestAnswer`.

Reviewer: ```suggestion
For example, the [Chainlink BTC/USD oracle ↗](https://data.chain.link/ethereum/mainnet/crypto-usd/btc-usd) has its JSON ABI [published on Etherscan ↗](https://etherscan.io/address/0xf4030086522a5beea4988f8ca5b36dbc97bee88c#code). When defining the data source spec, you can populate the `abi` field with the full ABI, and then set the `method` to `latestAnswer`.
```

:::note Shrinking the ABI
When populating the `abi` field on your data source spec, you can remove the methods and other fields that are not required by the oracle. We've done that in the sample data source above - only the `latestAnswer` method and its inputs and outputs are in the `abi` field.
:::

### Time trigger

> With Ethereum data source specs the validators will read the specified smart contract and method detailed in the ABI. The `trigger` instructs the validators when to do this. In the smple above, it will be called every 30 seconds.

Reviewer: ```suggestion
With Ethereum data source specs the validators will read the specified smart contract and method detailed in the ABI. The `trigger` instructs the validators when to do this. In the sample above, it will be called every 30 seconds.
```

### Normaliser

> A [JSONPath](https://datatracker.ietf.org/wg/jsonpath/about/) expression use to extract data from the JSON returned from the method call. In the spec above, `expression` is set to `$[0]`, which returns the first item in an array. `$` would return the complete result.

Reviewer: ```suggestion
A normaliser is a [JSONPath ↗](https://datatracker.ietf.org/wg/jsonpath/about/) expression used to extract data from the JSON returned from the method call. In the spec above, `expression` is set to `$[0]`, which returns the first item in an array. `$` would return the complete result.
```

## Open Oracle signed messages

> Vega's Data Sourcing framework supports signed ABI-encoded [Open Oracle ↗](https://github.com/compound-finance/open-oracle) or JSON messages. ABI-encoded signed messages can be verified to have come from the public key that signed them, which allows futures markets on Vega to use pricing data sourced from Ethereum.

Reviewer: ```suggestion
Vega's data sourcing framework supports signed ABI-encoded [Open Oracle ↗](https://github.com/compound-finance/open-oracle) or JSON messages. ABI-encoded signed messages can be verified to have come from the public key that signed them, which allows futures markets on Vega to use pricing data sourced from Ethereum.
```

> When it's time for a market to settle, someone needs to submit the data that matches the data source spec defined in the market. Any Vega keypair can submit the data. In the configuration for a market, a data source specification field dictates which data feeds it is interested in. In effect, it works as a filter. This specification means that the creator of an instrument for a market will choose in advance a price source, and which data fields the market requires to settle and terminate.

Reviewer: ```suggestion
When it's time for a market to settle, someone needs to submit the data that matches the data source spec defined in the market. Any Vega keypair can submit the data. In the configuration for a market, a data source specification field dictates which data feeds it is interested in.
```

## JSON signed message data
[JSON ↗](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON) messages are a simpler, more configurable alternative to Open Oracle data. They can be totally custom objects, as long as they are valid JSON. As they are not attested by any off-chain source in the way that Open Oracle messages are, and so it's generally advisable to check for an Open Oracle price source before choosing JSON data. The Vega key that signs the message will be referred to as the source for the data. 

> When it's time for a market to settle, someone needs to submit the data that matches the data source spec defined in the market. Any Vega keypair can submit the data. In the configuration for a market, a data source specification field dictates which data feeds it is interested in. In effect, it works as a filter. This specification means that the creator of an instrument for a market will choose in advance a price source, and which data fields the market requires to settle and terminate.

Reviewer: ```suggestion
When it's time for a market to settle, someone needs to submit the data that matches the data source spec defined in the market. Any Vega keypair can submit the data. In the configuration for a market, a data source specification field dictates which data feeds it is interested in.
```
